src/core/qgsinterval.cpp

```diff
@@ -212,14 +212,28 @@ QgsInterval QgsInterval::fromString( const QString &string )
 {
   double seconds = 0;
   const thread_local QRegularExpression rx( "([-+]?\\d*\\.?\\d+\\s+\\S+)", QRegularExpression::CaseInsensitiveOption );
+  const thread_local QRegularExpression rxtime( ".* (\\d{1,2}:){2}\\d{1,2}.*", QRegularExpression::CaseInsensitiveOption );
+
+  QRegularExpressionMatch matchtime = rxtime.match( string );
+  QString modedString = QString( string );
+  if ( matchtime.hasMatch() && std::count( modedString.begin(), modedString.end(), ':' ) == 2 ) //some part of the string contains 00:00:00 style duration
```

Comment:
aren't we guaranteed that the string has two ':' by the regular expression?

Reply:
Two is indeed the minimum, but I thought there might be cases where there might be more. Since it is a core class, I wanted to avoid messing around working code. 
I thought about it since then and I wonder if it would have been better to move that piece of code to the data provider ( src/providers/postgres/qgspostgresprovider.cpp@5064 ). 
Any thoughts?